### ISSUE TITLE
Anomalib インストール

### ISSUE THREAD

**Issue by KazukiMoriai** [KazukiMoriai/RustOne#5]
- Anomalib：複数の異常検知アルゴリズム（AutoEncoderを含む場合もある）を実装したライブラリ。
- 異常検知のパイプライン全体（データ読み込み、前処理、モデル学習、評価、可視化など）が整備されているため、データの送受信など包括的に利用可能と考える

**Comment by KazukiMoriai** [comment]
### **動作可能なもの**
1. anomalib: バージョン0.7.0がインストールされており、基本的な機能は使用可能です
2. numpy, scipy, matplotlib: 基本的な数値計算、科学計算、グラフ描画が可能
3. omegaconf, einops, kornia: 設定管理と画像処理の基本機能
4. OpenCV (headlessバージョン): 画像処理の基本機能（ただしGUIなし）
5. pytorch_lightning, timm: ディープラーニングのフレームワークとモデル
### **制限されている/インストールされていないもの**
1. imgaug: インストールされていないため、データ拡張（画像の回転、反転、ノイズ追加などの処理）機能が使用できません
2. av: インストールされていないため、動画処理機能が制限されています
3. GUI機能: OpenCVのheadlessバージョンを使用しているため、画像の表示などのGUI機能は使用できません
### 実際に可能なこと
* 異常検知の基本的なモデルのトレーニングと推論
* 画像の前処理と基本的な画像処理
* 結果の数値的な評価
### 制限されていること
* データ拡張を使った高度なトレーニング（imgaugがないため）
* 動画ファイルの処理（avがないため）
* リアルタイムの画像表示（headlessバージョンのため）
※anomalibの基本機能は使用できる状態だが、より高度なデータ拡張や動画処理などの機能を使いたい場合は、足りないパッケージを追加でインストールする必要あり
### メモ
**環境構築手順**（自動構築されない場合）
1. pip を最新版にアップグレード
`pip install --upgrade pip`
2. requirements.txtから厳密にインストール
`pip install -r requirements.txt`
3. ターミナルでバージョン確認
` pip list`
>Package                  Version
------------------------ -----------
aiohttp                  3.8.6
aiosignal                1.3.1
albumentations           1.3.1
anomalib                 0.7.0
antlr4-python3-runtime   4.9.3
async-timeout            4.0.3
asynctest                0.13.0
attrs                    24.2.0
av                       10.0.0
certifi                  2025.1.31
charset-normalizer       3.4.1
cycler                   0.11.0
docstring_parser         0.16
einops                   0.6.1
filelock                 3.12.2
fonttools                4.38.0
FrEIA                    0.2
frozenlist               1.3.3
fsspec                   2023.1.0
huggingface-hub          0.16.4
idna                     3.10
imageio                  2.31.2
imgaug                   0.4.0
importlib-metadata       6.7.0
importlib-resources      5.12.0
joblib                   1.3.2
jsonargparse             4.32.1
kiwisolver               1.4.5
kornia                   0.6.9
lightning-utilities      0.10.1
matplotlib               3.5.3
multidict                6.0.5
networkx                 2.6.3
numpy                    1.21.6
nvidia-cublas-cu11       11.10.3.66
nvidia-cuda-nvrtc-cu11   11.7.99
nvidia-cuda-runtime-cu11 11.7.99
nvidia-cudnn-cu11        8.5.0.96
omegaconf                2.3.0
opencv-python            4.11.0.86
opencv-python-headless   4.11.0.86
packaging                24.0
pandas                   1.3.5
Pillow                   9.5.0
pip                      24.0
pkg_resources            0.0.0
pyparsing                3.1.4
python-dateutil          2.9.0.post0
pytorch-lightning        1.9.5
pytz                     2025.1
PyWavelets               1.3.0
PyYAML                   6.0.1
qudida                   0.0.4
requests                 2.31.0
scikit-image             0.19.3
scikit-learn             1.0.2
scipy                    1.7.3

4. ターミナルで動作確認
 `python test_anomalib.py`
> anomalib バージョン: 0.7.0
anomalib は正常にインポートできました。基本的な機能は使用可能です。
主要な依存パッケージ (numpy, scipy, matplotlib) は使用可能です。
画像処理に必要な依存パッケージ (omegaconf, einops, kornia) は使用可能です。
OpenCV バージョン: 4.11.0
OpenCV-Python-Headless は使用可能です。
追加の依存パッケージ (pytorch-lightning, timm, imgaug) は使用可能です。
